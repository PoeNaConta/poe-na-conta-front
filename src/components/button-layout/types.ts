import { ReactNode } from 'react';

export interface ButtonLayoutProps {
  primaryButton: ReactNode;
  secondaryButton?: ReactNode;
  row?: boolean;
  column?: boolean;
  fullWidth?: boolean;
  space?: 'xxs' | 'xs' | 'sm' | 'md' | 'lg' | 'xl' | 'xxl';
}
