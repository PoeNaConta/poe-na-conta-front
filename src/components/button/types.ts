import { PropsWithChildren } from 'react';

export interface PrimaryButtonProps extends PropsWithChildren {
  primary: true;
  secondary?: never;
  onClick: () => void;
  fake?: never;
}

export interface SecondaryButtonProps extends PropsWithChildren {
  primary?: never;
  secondary: true;
  onClick: () => void;
  fake?: never;
}

export interface FakeButtonProps extends PropsWithChildren {
  primary?: boolean;
  secondary?: boolean;
  onClick?: undefined;
  fake: true;
}

export type ButtonProps =
  | FakeButtonProps
  | SecondaryButtonProps
  | PrimaryButtonProps;
