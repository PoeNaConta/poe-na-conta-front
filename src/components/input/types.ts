import { ChangeEvent, HTMLInputTypeAttribute } from 'react';

export interface InputProps {
  label?: string;
  id?: string;
  name?: string;
  type?: HTMLInputTypeAttribute;
  value?: string;
  placeholder?: string;
  onChange?: (event: ChangeEvent<HTMLInputElement>) => void;
}
