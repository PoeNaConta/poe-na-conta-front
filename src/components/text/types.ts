import { PropsWithChildren } from 'react';

export interface TextProps extends PropsWithChildren {
  as?: 'p' | 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h5' | 'span' | 'label';
  size?: 'xs' | 'sm' | 'md' | 'lg' | 'xl' | 'hero';
  weight?: 'light' | 'regular' | 'medium' | 'semibold' | 'bold';
  htmlFor?: string;
  primary?: boolean;
  secondary?: boolean;
  tertiary?: boolean;
  onBrand?: boolean;
  center?: boolean;
}
