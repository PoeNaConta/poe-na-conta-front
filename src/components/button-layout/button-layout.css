.lds--button-layout {
  display: flex;
}

.lds--button-layout--row {
  flex-direction: row;
}

.lds--button-layout--column {
  flex-direction: column;
}

.lds--button-layout--column > * {
  width: fit-content;
}

.lds--button-layout--full-width > * {
  flex-grow: 1;
  width: auto;
}

.lds--button-layout--spacing--xxs {
  gap: var(--lds-spacing-xxs);
}

.lds--button-layout--spacing--xs {
  gap: var(--lds-spacing-xs);
}

.lds--button-layout--spacing--sm {
  gap: var(--lds-spacing-sm);
}

.lds--button-layout--spacing--md {
  gap: var(--lds-spacing-md);
}

.lds--button-layout--spacing--lg {
  gap: var(--lds-spacing-lg);
}

.lds--button-layout--spacing--xl {
  gap: var(--lds-spacing-xl);
}

.lds--button-layout--spacing--xxl {
  gap: var(--lds-spacing-xxl);
}

.lds--button-layout--spacing--xxs {
  gap: var(--lds-spacing-xxs);
}
