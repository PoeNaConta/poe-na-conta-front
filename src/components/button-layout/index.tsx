import './button-layout.css';
import { ButtonLayoutProps } from './types';

export default function ButtonLayout({
  primaryButton,
  secondaryButton,
  row,
  column,
  fullWidth = true,
}: ButtonLayoutProps) {
  const base = 'lds--button-layout';

  const getDirectionModifier = () => {
    if (row) return `${base}--row`;
    if (column) return `${base}--column`;
    return '';
  };

  const getFullWidthModifier = () => {
    if (fullWidth) return `${base}--full-width`;
    return '';
  };

  const className = `${base} ${getDirectionModifier()} ${getFullWidthModifier()}`;
  return (
    <div className={className}>
      {primaryButton}
      {secondaryButton ? secondaryButton : null}
    </div>
  );
}
