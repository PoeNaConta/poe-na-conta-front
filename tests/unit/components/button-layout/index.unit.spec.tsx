import React from 'react';
import ButtonLayout from '@components/button-layout';
import { render } from '@testing-library/react';
import { ButtonLayoutProps } from '@components/button-layout/types';

describe('ButtonLayout', () => {
  it('should render button layout with primary button', () => {
    const { container } = render(
      <ButtonLayout primaryButton={<button>example primary button</button>} />,
    );

    expect(container).toMatchSnapshot();
  });

  it('should render button layout with secondary button', () => {
    const { container } = render(
      <ButtonLayout
        primaryButton={<button>example primary button</button>}
        secondaryButton={<button>example secondary button</button>}
      />,
    );

    expect(container).toMatchSnapshot();
  });

  it.each([{ row: true }, { column: true }])(
    'should render button layout with direction %s',
    (
      directionProps:
        | { row: true; column?: never }
        | { row?: never; column: true },
    ) => {
      const { container } = render(
        <ButtonLayout
          {...directionProps}
          primaryButton={<button>example primary button</button>}
        />,
      );

      expect(container).toMatchSnapshot();
    },
  );

  it.each(['xxs', 'xs', 'sm', 'md', 'lg', 'xl', 'xxl'])(
    'should render button layout with direction %s',
    (space: ButtonLayoutProps['space']) => {
      const { container } = render(
        <ButtonLayout
          space={space}
          primaryButton={<button>example primary button</button>}
          secondaryButton={<button>example secondary button</button>}
        />,
      );

      expect(container).toMatchSnapshot();
    },
  );

  it('should not render button layout as row', () => {
    const { container } = render(
      <ButtonLayout
        row={false}
        primaryButton={<button>example primary button</button>}
      />,
    );

    expect(container.firstElementChild).not.toHaveClass(
      'lds--button-layout--row',
    );
  });

  it.each([true, false])(
    'should render button layout with fullWidth=%s',
    (fullWidth: boolean) => {
      const { container } = render(
        <ButtonLayout
          fullWidth={fullWidth}
          primaryButton={<button>example primary button</button>}
        />,
      );

      expect(container).toMatchSnapshot();
    },
  );
});
