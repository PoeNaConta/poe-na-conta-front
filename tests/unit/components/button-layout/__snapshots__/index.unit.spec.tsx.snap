// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ButtonLayout > should render button layout with direction { column: true } 1`] = `
<div>
  <div
    class="lds--button-layout lds--button-layout--column lds--button-layout--full-width lds--button-layout--spacing--xs"
  >
    <button>
      example primary button
    </button>
  </div>
</div>
`;

exports[`ButtonLayout > should render button layout with direction { row: true } 1`] = `
<div>
  <div
    class="lds--button-layout lds--button-layout--row lds--button-layout--full-width lds--button-layout--spacing--xs"
  >
    <button>
      example primary button
    </button>
  </div>
</div>
`;

exports[`ButtonLayout > should render button layout with direction lg 1`] = `
<div>
  <div
    class="lds--button-layout  lds--button-layout--full-width lds--button-layout--spacing--lg"
  >
    <button>
      example primary button
    </button>
    <button>
      example secondary button
    </button>
  </div>
</div>
`;

exports[`ButtonLayout > should render button layout with direction md 1`] = `
<div>
  <div
    class="lds--button-layout  lds--button-layout--full-width lds--button-layout--spacing--md"
  >
    <button>
      example primary button
    </button>
    <button>
      example secondary button
    </button>
  </div>
</div>
`;

exports[`ButtonLayout > should render button layout with direction sm 1`] = `
<div>
  <div
    class="lds--button-layout  lds--button-layout--full-width lds--button-layout--spacing--sm"
  >
    <button>
      example primary button
    </button>
    <button>
      example secondary button
    </button>
  </div>
</div>
`;

exports[`ButtonLayout > should render button layout with direction xl 1`] = `
<div>
  <div
    class="lds--button-layout  lds--button-layout--full-width lds--button-layout--spacing--xl"
  >
    <button>
      example primary button
    </button>
    <button>
      example secondary button
    </button>
  </div>
</div>
`;

exports[`ButtonLayout > should render button layout with direction xs 1`] = `
<div>
  <div
    class="lds--button-layout  lds--button-layout--full-width lds--button-layout--spacing--xs"
  >
    <button>
      example primary button
    </button>
    <button>
      example secondary button
    </button>
  </div>
</div>
`;

exports[`ButtonLayout > should render button layout with direction xxl 1`] = `
<div>
  <div
    class="lds--button-layout  lds--button-layout--full-width lds--button-layout--spacing--xxl"
  >
    <button>
      example primary button
    </button>
    <button>
      example secondary button
    </button>
  </div>
</div>
`;

exports[`ButtonLayout > should render button layout with direction xxs 1`] = `
<div>
  <div
    class="lds--button-layout  lds--button-layout--full-width lds--button-layout--spacing--xxs"
  >
    <button>
      example primary button
    </button>
    <button>
      example secondary button
    </button>
  </div>
</div>
`;

exports[`ButtonLayout > should render button layout with fullWidth=false 1`] = `
<div>
  <div
    class="lds--button-layout   lds--button-layout--spacing--xs"
  >
    <button>
      example primary button
    </button>
  </div>
</div>
`;

exports[`ButtonLayout > should render button layout with fullWidth=true 1`] = `
<div>
  <div
    class="lds--button-layout  lds--button-layout--full-width lds--button-layout--spacing--xs"
  >
    <button>
      example primary button
    </button>
  </div>
</div>
`;

exports[`ButtonLayout > should render button layout with primary button 1`] = `
<div>
  <div
    class="lds--button-layout  lds--button-layout--full-width lds--button-layout--spacing--xs"
  >
    <button>
      example primary button
    </button>
  </div>
</div>
`;

exports[`ButtonLayout > should render button layout with secondary button 1`] = `
<div>
  <div
    class="lds--button-layout  lds--button-layout--full-width lds--button-layout--spacing--xs"
  >
    <button>
      example primary button
    </button>
    <button>
      example secondary button
    </button>
  </div>
</div>
`;
